### DISCUSSION TITLE
useResetAtom not working (typescript complained error ) if a atom is created from withImmer

### DISCUSSION THREAD

**Discussion by lilong7676** [pmndrs/jotai#2621] [Bug report]
a minimal reproduction:

https://codesandbox.io/p/sandbox/jotai-demo-xp8m96?file=%2Fpackage.json

<img width="909" alt="image" src="https://github.com/pmndrs/jotai/assets/21004230/81b4dc8d-98b7-4e5d-8000-5360d14e7062">

**Comment by dai-shi** [comment]
Thanks for reporting. Maybe our docs are not well, but useResetAtom only works for resettable atoms, and `withImmer` doesn't support it. I think it's possible to make `withImmer` so that it keeps resettable capability. Would anyone like to give it a try?

**Reply by tien** [reply]
Hopefully this will fix it jotaijs/jotai-immer#11 🙏
